File: bridge/c-sharp/EmailMessageSender.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Bridge
{
    // UML diyagramındaki ConcreteImplementor (EmailMessageSender) kısmına denk gelmektedir.
    // Implementor arayüzünü (IMessageSender) uygular.
    public class EmailMessageSender : IMessageSender
    {
        public void SendMessage(Message message)
        {
            Console.WriteLine("EmailMessageSender: Sending email message...");
        }
    }
}

File: bridge/typescript/email-message-sender.ts
import {IMessageSender} from "./message-sender.interface";
import {Message} from "./message";

// UML diyagramındaki ConcreteImplementor (EmailMessageSender) kısmına denk gelmektedir.
// Implementor arayüzünü (IMessageSender) uygular.
export class EmailMessageSender implements IMessageSender {
  public sendMessage(message: Message): void {
    console.log('EmailMessageSender: Sending email message...');
  }
}
